### PR TITLE
#2817 Rotation tool: cancel rotation by pressing 'right click' button

### PR DESCRIPTION
**Pull request by Inga-Vishnivetskaia**
## How the feature works? / How did you fix the issue?

- the rotation should is cancelled and the structure revert to the original position it was in when rotation was started
- the structure stays selected

[video: Cancel Rotate on right click](https://clipchamp.com/watch/ymYauyz09bW)

## Check list
- [ ] unit-tests written
- [x] e2e-tests written
- [x] documentation updated
- [x] PR name follows the pattern `#1234 – issue name`
- [x] branch name doesn't contain '#'
- [x] PR is linked with the issue
- [x] base branch (master or release/xx) is correct

### CODE REVIEW

File: packages/ketcher-react/src/script/ui/state/mouse.ts [hunk 1]
```diff
@@ -33,5 +33,24 @@ export function initMouseListener(element) {
     element.addEventListener('pointermove', (event: MouseEvent) =>
       throttledHandleMouseMove(dispatch, event)
     )
+    element.addEventListener(
+      'mousedown',
+      function (event: MouseEvent) {
+        const isBothLeftAndRightClick = event.buttons === 3
+        if (isBothLeftAndRightClick) {
+          rightClickHandle(getState)
```
Comment (yuleicul) [inline packages/ketcher-react/src/script/ui/state/mouse.ts:41]: I'd suggest naming function starting with verb, how about `handleRightClick`

File: packages/ketcher-react/src/script/ui/views/components/ContextMenu/ContextMenuTrigger.tsx [hunk 2]
```diff
@@ -81,7 +81,8 @@ const ContextMenuTrigger: React.FC<PropsWithChildren> = ({ children }) => {
       let triggerType: ContextMenuTriggerType
 
       if (!closestItem) {
-        if (selection) {
+        const isLeftMouseButtonPressed = event.buttons === 1
+        if (selection && !isLeftMouseButtonPressed) {
```
Comment (yuleicul) [inline packages/ketcher-react/src/script/ui/views/components/ContextMenu/ContextMenuTrigger.tsx:85]: It takes a while to think why `isLeftMouseButtonPressed` is needed. Maybe we can:
```suggestion
        const isLeftMouseButtonPressed = event.buttons === 1
        const isRotationReverted = isLeftMouseButtonPressed
        if (selection && !isRotationReverted) {
```

File: packages/ketcher-react/src/script/ui/state/mouse.ts [hunk 3]
```diff
@@ -33,5 +33,24 @@ export function initMouseListener(element) {
...
       throttledHandleMouseMove(dispatch, event),
     );
+    element.addEventListener(
+      'mousedown',
+      function (event: MouseEvent) {
+        const isBothLeftAndRightClick = event.buttons === 3;
+        if (isBothLeftAndRightClick) {
+          rightClickHandle(getState);
+        }
+      },
```
Comment (yuleicul) [inline packages/ketcher-react/src/script/ui/state/mouse.ts:43]: Please remember to remove this listener

File: ketcher-autotests/tests/Structure-Creating-&-Editing/Actions-With-Structures/Rotation/rotation.spec.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+  test.afterEach(async ({ page }) => {
+    await takeEditorScreenshot(page);
+  });
+
+  test('Cancel rotation on right click', async ({ page }) => {
+    const editor = await page.locator('[class*="App-module_canvas"]');
+
+    // 1. take snapshot of current structure in editor
+    const snapshotBeforeRotation = await editor.screenshot();
```
Comment (Nitvex) [inline ketcher-autotests/tests/Structure-Creating-&-Editing/Actions-With-Structures/Rotation/rotation.spec.ts:29]: for this we have a helper `takeEditorScreenshot`

File: ketcher-autotests/tests/utils/clicks/selectButtonByTitle.ts [hunk 5]
```diff
@@ -5,7 +5,10 @@ export async function selectButtonByTitle(title: ToolbarButton, page: Page) {
   await page.locator(`button[title*="${title}"]`).click();
 }
 
-// export async function getRotationHandleCoordinates(page: Page): Promise<Point> {
-//   // @ts-ignore
-//   return await page.evaluate(() => window.ketcher.editor.rotateController.handleCenter);
-// }
+export async function getRotationHandleCoordinates(page: Page) {
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
```
Comment (Nitvex) [inline ketcher-autotests/tests/utils/clicks/selectButtonByTitle.ts:9]: Can we fix this warning?
Previously, we had two different repos and it wasn't easy to make changes simultaneously in both repos. But errors like that should be easily fixed now.

File: ketcher-autotests/tests/Structure-Creating-&-Editing/Actions-With-Structures/Rotation/rotation.spec.ts [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  };
+
+  test.beforeEach(async ({ page }) => {
+    await page.goto('');
+    await drawBenzeneRing(page);
+    await selectSingleBondTool(page);
+  });
+
+  test('Cancel rotation on right click', async ({ page }) => {
+    const snapshotBeforeRotation = await getEditorScreenshot(page);
```
Comment (Nitvex) [inline ketcher-autotests/tests/Structure-Creating-&-Editing/Actions-With-Structures/Rotation/rotation.spec.ts:22]: I don't understand this test. First of all, why do you need to choose `selectSingleBondTool`?
Secondly, you do literally nothing with the structure. 
1) You get coordinates of rotation handle
2) Then you click on it – no action is performed on the structure
3) Then you move mouse to some coordinates (may be worth renaming them to something like `COORDINATES_TO_PERFORM_ROTATION`)
4) And then make a right click.

This way, test is supposed to be:
1) You get coordinates of rotation handle.
2) `mousedown` on rotation handle, not click
3) Move mouse to some coordinates (you need to drag handle, not simply click on it)
4) Press right click

File: ketcher-autotests/tests/Structure-Creating-&-Editing/Actions-With-Structures/Rotation/rotation.spec.ts [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+    await page.mouse.click(x, y);
+
+    await page.mouse.move(MOUSE_COORDS.x, MOUSE_COORDS.y);
+
+    await page.mouse.click(MOUSE_COORDS.x, MOUSE_COORDS.y, { button: 'right' });
+
+    const snapshotAfterRotation = await getEditorScreenshot(page);
+
+    expect(snapshotBeforeRotation).toEqual(snapshotAfterRotation);
```
Comment (Nitvex) [inline ketcher-autotests/tests/Structure-Creating-&-Editing/Actions-With-Structures/Rotation/rotation.spec.ts:36]: Could you figure out, why did we get an empty screenshot?

**Review by Nitvex — CHANGES_REQUESTED** [review]

**Review by AleksandraSmolianinova — APPROVED** [review]
Previewed. No cooments

**Review by Nitvex — APPROVED** [review]

**Pull request merged**